openmdao/visualization/n2_viewer/n2_viewer.py

```diff
@@ -77,73 +77,87 @@ def _convert_ndarray_to_support_nans_in_json(val):
     object : list, possibly nested
         The equivalent list with any nan values replaced with the string "nan".
     """
+    # do a quick check for any nans or infs and if not we can avoid the slow check
+    nans = np.where(np.isnan(val))
+    infs = np.where(np.isinf(val))
+    if nans[0].size == 0 and infs[0].size == 0:
+        return val.tolist()
+
     val_as_list = val.tolist()
     _convert_nans_in_nested_list(val_as_list)
-    return(val_as_list)
-
-
-def _get_var_dict(system, typ, name, is_parallel):
-    if name in system._var_discrete[typ]:
-        meta = system._var_discrete[typ][name]
-        is_discrete = True
-    else:
-        if name in system._var_abs2meta['output']:
-            meta = system._var_abs2meta['output'][name]
-        else:
-            meta = system._var_abs2meta['input'][name]
-        name = system._var_abs2prom[typ][name]
-        is_discrete = False
-
-    var_dict = {}
-
-    var_dict['name'] = name
-    var_dict['type'] = typ
-    if typ == 'output':
-        isimplicit = isinstance(system, ImplicitComponent)
-        var_dict['type'] = 'output'
-        var_dict['implicit'] = isimplicit
+    return val_as_list
+
+
+def _get_var_dict(system, typ, name, is_parallel, is_implicit):
+    if name in system._var_abs2meta[typ]:
+        meta = system._var_abs2meta[typ][name]
+        prom = system._var_abs2prom[typ][name]
+        val = meta['val']
+        is_dist = MPI is not None and meta['distributed']
+
+        var_dict = {
+            'name': prom,
+            'type': typ,
+            'dtype': type(val).__name__,
+            'is_discrete': False,
+            'distributed': is_dist,
+            'shape': str(meta['shape']),
+        }
+
+        if typ == 'output':
+            var_dict['implicit'] = is_implicit
+            vec = system._outputs
+        else:  # input
```

Comment:
could we get a code comment here explaining the `if MPI` ?

Reply:
done